pkg/egressservice/egressservice_test.go

```diff
@@ -0,0 +1,601 @@
+			nsname:        egressTestSvcNamespace,
+			sourceIPBy:    "LoadBalancerIP",
+			newVrfNetwork: " ",
+			nodeSelector:  map[string]string{},
+			expectedError: fmt.Errorf("Cannot use empty VRF network"),
+		},
+	}
+
+	t.Logf("=== Starting Update tests ===\n")
+
+	for _, testCase := range testCases {
+		testEgressServiceBuilder, err := Pull(buildTestClientWithDummyEgressService(),
+			testCase.name, testCase.nsname)
+
+		assert.Nil(t, err)
+
+		if testCase.newVrfNetwork != "" {
+			t.Logf("\t=> Updating EgressService's Network parameter\n")
+
+			testEgressServiceBuilder = testEgressServiceBuilder.WithVRFNetwork(testCase.newVrfNetwork)
+		}
+
+		if len(testCase.nodeSelector) != 0 {
+			t.Logf("\t=> Updating EgressService's nodeSelector parameter\n")
+
+			testEgressServiceBuilder = testEgressServiceBuilder.WithNodeLabelSelector(testCase.nodeSelector)
+		}
+
+		t.Logf("\t=> Updating EgressService\n")
+
+		testEgressServiceBuilder, err = testEgressServiceBuilder.Update()
+
+		t.Logf("\tError during update: %v\n", err)
+
+		if testCase.expectedError != nil {
+			t.Logf("\t\t==> Asserting error handling\n")
+			assert.Equal(t, testCase.expectedError, err)
+		} else {
+			assert.Equal(t, testCase.name, testEgressServiceBuilder.Object.Name)
+			assert.Equal(t, testCase.nsname, testEgressServiceBuilder.Object.Namespace)
+
+			t.Logf("\t\t==> Updated EgressService Definition\n%#v\n", testEgressServiceBuilder.Definition)
+			t.Logf("\t\t==> Updated EgressService Object\n%#v\n", testEgressServiceBuilder.Object)
+
+			if testCase.newVrfNetwork != "" {
+				t.Logf("\t\t==> Asserting Network was updated\n")
+				assert.Equal(t, testCase.newVrfNetwork, testEgressServiceBuilder.Object.Spec.Network)
+				assert.Equal(t,
+					testEgressServiceBuilder.Definition.Spec.Network,
+					testEgressServiceBuilder.Object.Spec.Network)
+			}
+
+			if len(testCase.nodeSelector) != 0 {
+				t.Logf("\t\t===> Asserting NodeSelector was updated\n")
+
+				for key, val := range testCase.nodeSelector {
+					t.Logf("\t\t===> Processing: %q -> %q", key, val)
+
+					assert.Equal(t,
+						testEgressServiceBuilder.Definition.Spec.NodeSelector.MatchLabels[key],
+						val)
+
+					assert.Equal(t,
+						testEgressServiceBuilder.Object.Spec.NodeSelector.MatchLabels[key],
+						val)
+				}
+			}
+		}
+	}
+}
+
+func TestCreate(t *testing.T) {
+	testCases := []struct {
+		name          string
+		namespace     string
+		sourceIPBy    string
+		expectedError error
+	}{
+		{
+			name:          egressTestSvcName,
+			namespace:     egressTestSvcNamespace,
+			sourceIPBy:    "Network",
+			expectedError: nil,
+		},
+		{
+			name:          egressTestSvcName,
+			namespace:     egressTestSvcNamespace,
+			sourceIPBy:    "LoadBalancerIP",
+			expectedError: nil,
+		},
+		{
+			name:          egressTestSvcName,
+			namespace:     egressTestSvcNamespace,
+			sourceIPBy:    "fake",
+			expectedError: fmt.Errorf("Invalid sourceIPBy parameter for the EgressService"),
```

Comment:
correct, but `Create` method calls _validate_ which prevents method from creating an object if definition is wrong so I don't see why it shouldn't be here

Reply:
@yprokule True, I think it's good to test the fact that validate is called. But since it's called for every method, it'd be a lot to verify all of the ways it can fail in all tests. [Other tests](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/mco/kubeletconfig_test.go#L170) tend to have one thing they change to verify validate. Then all of the specific messages should come from `NewEgressServiceBuilder` or `With***` so having them there is enough

Reply:
@klaskosk I'm leaving one instance for the _validate_ coverage and removing the rest, is that OK?